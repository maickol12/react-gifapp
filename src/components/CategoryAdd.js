import Reat, { useState } from 'react';
import PropTypes from 'prop-types';


const CategoryAdd = ({ setCategory }) => {
    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }
    const handleSubmit = (e) => {
        e.preventDefault();
        if( inputValue.trim().length > 2 ){
            setCategory( value => [inputValue, ...value]);
            setInputValue('')
        }
    }
    return(
        <div>
            <form onSubmit={handleSubmit}>
                <p>{ inputValue }</p>
                <input
                    type="text"
                    value={ inputValue }
                    onChange={ handleInputChange } />
                
            </form>
        </div>
    )
}

CategoryAdd.propTypes = {
    setCategory: PropTypes.func.isRequired
}

export default CategoryAdd;