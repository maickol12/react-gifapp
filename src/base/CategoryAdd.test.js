import { shallow } from 'enzyme';
import CategoryAdd from '../components/CategoryAdd';
import '@testing-library/jest-dom';
describe('Prueba eb <CategoryAdd />',() => {

    const setCategory = jest.fn();
    let wrapper;

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow( <CategoryAdd setCategory={ setCategory } /> );
    });


    test('debe de mostrarse correctamente',() => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe de cambiar la caja de texto',() => {
        const input = wrapper.find('input');
        const value = 'Hola Mundo';
        input.simulate('change', {target:{ value } });

        expect( wrapper.find('p').text().trim() ).toBe( value )
    });

    test('No debe de postear la informaciÃ³n con submit ',() => {
        wrapper.find('form').simulate('submit',{ preventDefault(){}});

        expect( setCategory ).not.toHaveBeenCalled();
    });

    test('debe de llamar el set category y limpiar la caja de texto',() => {
        const value = 'Hola Galaxia';
        wrapper.find('input').simulate('change',{target:{value}});
        wrapper.find('form').simulate('submit',{ preventDefault(){}})
        expect( setCategory ).toHaveBeenCalled( );
        expect( setCategory ).toHaveBeenCalledTimes( 1 );
        expect( setCategory ).toHaveBeenCalledWith( expect.any(Function) );
        expect( wrapper.find('input').prop('value') ).toBe('');
    });
});