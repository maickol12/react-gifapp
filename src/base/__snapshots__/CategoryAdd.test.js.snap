// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba eb <CategoryAdd /> debe de mostrarse correctamente 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <p />
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
</div>
`;
